node/core/chain-selection/src/db_backend/v1.rs

```diff
@@ -240,7 +241,9 @@ impl Backend for DbBackend {
 					_ => None,
 				}
 			})
-			.take_while(|(at, _)| *at <= up_to.into())
+			.enumerate()
+			.take_while(|(idx, (at, _))| *at <= up_to.into() && *idx < max_elements)
```

Comment:
I think we could loose some elements if the timestamps are not sorted because we use `idx` instead as a counter.

Reply:
But the timestamps are guaranteed to be sorted, as otherwise the whole code would be incorrect.